src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts

```diff
@@ -17,4 +25,34 @@ export class AchievementEarnedPopupComponent {
         this.achievement = this.navParams.get('achievement');
         this.earnedAwards = this.navParams.get('earnedAwards');
     }
+
+    public async ngOnInit() {
+
+        const url = this.achievement.meta.icon || BASE_ICON;
+
+        try {
+            const res = await fetch(url, { method: 'HEAD' });
+            const newPrimaryColor = res.headers.get('x-goog-meta-primarycolor');
+            if(newPrimaryColor) {
+                this.baseColor = newPrimaryColor;
+            }
+        // unable to set color here
+        } catch(e) {}
+    }
+
+    public get boxShadow(): string {
+        const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(this.baseColor);
+        const rgb = result ? {
+            r: parseInt(result[1], 16),
+            g: parseInt(result[2], 16),
+            b: parseInt(result[3], 16)
+        } : null;
+
+        const boxShadow = '0px 23px 18px';
+        if(!rgb) {
+            return `${boxShadow} rgba(241, 32, 73, 0.29)`;
```

Comment:
can you add a comment or constant describing what `241, 32, 73, 0.29` is?

Reply:
can do!